src/endpoint/s3/s3_rest.js

```diff
@@ -266,7 +265,7 @@ function _get_method_from_req(req) {
         dbg.error(`Got a not supported S3 op ${req.op_name} - doesn't suppose to happen`);
         throw new S3Error(S3Error.InternalError);
     }
-    if (req.query && req.query.versionId && s3_op.versioned) {
+    if (s3_utils.parse_version_id(req.query?.versionId) && s3_op.versioned) {
```

Comment:
again also here, the actual value of version id is not needed and should not be validated. the check was only if version id is set for the request. if you want you can add a separate `s3_utils.is_versioning_request(req)` which will just return `Boolean(req.query?.versionId)`